silkworm/silkrpc/core/rawdb/chain.cpp

```diff
@@ -172,14 +172,13 @@ Task<Receipts> read_raw_receipts(const DatabaseReader& reader, BlockNum block_nu
     const auto block_key = silkworm::db::block_key(block_number);
     const auto data = co_await reader.get_one(db::table::kBlockReceiptsName, block_key);
     SILK_TRACE << "read_raw_receipts data: " << silkworm::to_hex(data);
+    Receipts receipts{};
     if (data.empty()) {
-        co_return Receipts{};  // TODO(canepat): use std::null_opt with Task<std::optional<Receipts>>?
+        co_return receipts;
     }
-    Receipts receipts{};
     const bool decoding_ok{cbor_decode(data, receipts)};
     if (!decoding_ok) {
-        SILK_WARN << "cannot decode raw receipts in block: " << block_number;
-        co_return receipts;
+        throw std::runtime_error("cannot decode raw receipts in block: " + std::to_string(block_number));
     }
     SILK_DEBUG << "#receipts: " << receipts.size();
 
```

Comment:
You can add here an early return when there are zero receipts to avoid the next `walk` call.